### ISSUE TITLE
[bitnami/posgresql-ha] Why include the cluster domain in the cluster node hosts? And what about the namespace?

### ISSUE THREAD

**Issue by JeroenVanOort** [bitnami/charts#4923]
**Which chart**:
postgresql-ha

**Is your feature request related to a problem? Please describe.**
I'm having problems with the length limit on the cluster node hosts, as defined [here](https://github.com/bitnami/charts/blob/master/bitnami/postgresql-ha/templates/postgresql/statefulset.yaml#L225) and validated [here](https://github.com/bitnami/charts/blob/master/bitnami/postgresql-ha/templates/_helpers.tpl#L563). I'm frequently confronted with the error message saying that the name is too long.

I wonder why the cluster domain is included in the host names. As far as I know, Kubernetes supports shortened names, e.g. `foo.namespace` instead of `foo.namespace.svc.cluster.local`. Why not use them?

And if the cluster domain can be omitted, would be possible to omit the namespace too? It would only need to be there when the namespaces for postgres and pgpool differ.

**Describe the solution you'd like**
If possible, I would suggest to omit the cluster domain and thereby increasing the headroom for the release name and namespace.

And if that's possible, we could possibly omit the namespace too.

**Describe alternatives you've considered**
I've been shortening release names and I've added an alias (pgha) for this chart in order to make the name less lengthy.

**Additional context**
I'm using this chart as a dependency of an application. The application can be deployed into a review environment, allowing the customer to see the changes we've made. All reviews are deployed to their own namespace, which name included the project name. This leads to hostnames like these: `deployment-pgha-postgresql-00.deployment-pgha-postgresql-headless.project-name-identifier-review-deployment-tdu9h8.svc.cluster.local:1234`.

**Comment by juan131** [comment]
Hi @JeroenVanOort 

> I wonder why the cluster domain is included in the host names. As far as I know, Kubernetes supports shortened names, e.g. foo.namespace instead of foo.namespace.svc.cluster.local. Why not use them?

It's a matter of compatibility between clusters that may have a different cluster domain.

I recommend you using the `nameOverride ` or `fullnameOverride` parameter to avoid having such long names. For instance:

```console
helm install first-postgresql-ha bitnami/postgresql-ha --set fullnameOverride=pgha-1
```

**Comment by JeroenVanOort** [comment]
Thank you for your reply.

I understand that there are clusters that have a non-default domain. But that doesn't matter when the domain is no longer used in the host, right? If I understand correctly, `podname.servicename.namespace` is an alias for `podname.servicename.namespace.svc.clusterdomain`.

If I understand correctly, this is what is used in the [redis-cluster chart](https://github.com/bitnami/charts/blob/master/bitnami/redis-cluster/templates/redis-statefulset.yaml#L125).

**Comment by juan131** [comment]
As it's explained at [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service), when using a headless service to expose the SRV Records created for each pod has the form:

- `pod-name.headless-svc.namespace.svc.cluster-domain.example`

To be honest, I'm not 100% sure why we use the FQDN instead of using one of the sorter aliases. I'll have to investigate the reasons behind this, but it's highly likely to be related with certain DNS add-ons that don't generate the aliases by default.

**Comment by jon-nfc** [comment]
This issue needs to be re-opened.  Why everyone uses a bot to close issues that haven't been resolved is beyond me.

I hit a bug today due to the fact that the full domain was used with hostname. I changed the cluster domain name and whilst everything else was fine after convergence had occurred, postgres failed to start. In every place of the postgress manifests that referenced the domain, were updated. However something still in the postgres container was still using the old cluster domain.


> As it's explained at [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service), when using a headless service to expose the SRV Records created for each pod has the form:
> 
> * `pod-name.headless-svc.namespace.svc.cluster-domain.example`
> 
> To be honest, I'm not 100% sure why we use the FQDN instead of using one of the sorter aliases. I'll have to investigate the reasons behind this, but it's highly likely to be related with certain DNS add-ons that don't generate the aliases by default.

this is clearly a case of monkey theory (monkey see, monkey do) as to why something has been done. This has subsequently been the cause of the bug I encountered.

First things first, they are not 'aliases'. With the exception of the first item, that's the host name, the rest to the left of the TLD are sub-domains. Within an established networking environment DHCP has a option (119) to pass search names. on windows this is done via group policy and within kubernetes /etc/resolv.conf is updated to include `search {space separated list of domains}` .

There is only a single downside that i can see with using DNS domain search. Although I'm not concerned as the benefits outweigh the downsides. for each domain that exists and doesn't return a result another DNS lookup is conducted.  The benefits? easily move items around kubernetes without having to update DNS.

Bottom line, DNS provides an easy to use method to connect to a host where layer 3 networking might change and as such kubernetes like most networking infrastructure uses DNS name searching and anywhere where the full domain name along with hostname is used, must be removed.

**Comment by carrodher** [comment]
Thank you for bringing this issue to our attention. We appreciate your involvement! If you're interested in contributing a solution, we welcome you to create a pull request. The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).

Your contribution will greatly benefit the community. Feel free to reach out if you have any questions or need assistance.

**Comment by jon-nfc** [comment]
> Thank you for bringing this issue to our attention. We appreciate your involvement! If you're interested in contributing a solution, we welcome you to create a pull request. The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).
> 
> Your contribution will greatly benefit the community. Feel free to reach out if you have any questions or need assistance.

thanks bot!!
